src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareIncludeExcludeAllNodesOperation.cs

```diff
@@ -0,0 +1,120 @@
+namespace Microsoft.SqlTools.ServiceLayer.SchemaCompare
+{
+    /// <summary>
+    /// Class to represent an in-progress schema compare include/exclude all Node operation
+    /// </summary>
+    class SchemaCompareIncludeExcludeAllNodesOperation : ITaskOperation
+    {
+        private CancellationTokenSource cancellation = new CancellationTokenSource();
+        private bool disposed = false;
+
+        /// <summary>
+        /// Gets the unique id associated with this instance.
+        /// </summary>
+        public string OperationId { get; private set; }
+
+        public SchemaCompareIncludeExcludeAllNodesParams Parameters { get; }
+
+        protected CancellationToken CancellationToken { get { return this.cancellation.Token; } }
+
+        public string ErrorMessage { get; set; }
+
+        public SqlTask SqlTask { get; set; }
+
+        public SchemaComparisonResult ComparisonResult { get; set; }
+
+        public bool Success { get; set; }
+
+        public List<DiffEntry> AllIncludedOrExcludedDifferences;
+
+
+        public SchemaCompareIncludeExcludeAllNodesOperation(SchemaCompareIncludeExcludeAllNodesParams parameters, SchemaComparisonResult comparisonResult)
+        {
+            Validate.IsNotNull("parameters", parameters);
+            this.Parameters = parameters;
+            Validate.IsNotNull("comparisonResult", comparisonResult);
+            this.ComparisonResult = comparisonResult;
+        }
+
+        /// <summary>
+        /// Execute will include/exclude all differences in the schema compare result.
+        /// </summary>
+        /// <param name="mode"></param>
+        public void Execute(TaskExecutionMode mode)
+        {
+            this.CancellationToken.ThrowIfCancellationRequested();
+
+            try
+            {
+                var schemaDifferences = new List<SchemaDifference>(this.ComparisonResult.Differences);
+                this.IncludeExcludeAllDifferences(schemaDifferences);
+            }
+            catch (Exception e)
+            {
+                ErrorMessage = e.Message;
+                Logger.Error(string.Format("Schema compare includ/exclude all operation {0} failed with exception {1}", this.OperationId, e.Message));
+                throw;
+            }
+
+            this.AllIncludedOrExcludedDifferences = new List<DiffEntry>();
+            if (this.ComparisonResult.Differences != null)
+            {
+                foreach (SchemaDifference difference in this.ComparisonResult.Differences)
+                {
+                    DiffEntry diffEntry = SchemaCompareUtils.CreateDiffEntry(difference, null, this.ComparisonResult);
+                    this.AllIncludedOrExcludedDifferences.Add(diffEntry);
+                }
+            }
+        }
+
+        private void IncludeExcludeAllDifferences(List<SchemaDifference> schemaDifferences)
+        {
+            var problematicDifferences = new List<SchemaDifference>();
+            foreach (var difference in schemaDifferences)
```

Comment:
use `SchemaDifference` instead of `var` if the type isn't already on the same line so readers don't have to hunt for the type.

Reply:
Sounds good, but the type is present on line 84 where the List is passed in to the method.